engine/collection/compliance/engine.go

```diff
@@ -166,23 +154,12 @@ func (e *Engine) OnSyncedClusterBlock(syncedBlock flow.Slashable[*messages.Clust
 	}
 }
 
-// finalizationProcessingLoop is a separate goroutine that performs processing of finalization events
-func (e *Engine) finalizationProcessingLoop(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
-	ready()
-
-	doneSignal := ctx.Done()
-	blockFinalizedSignal := e.finalizedBlockNotifier.Channel()
-	for {
-		select {
-		case <-doneSignal:
-			return
-		case <-blockFinalizedSignal:
-			// retrieve the latest finalized header, so we know the height
-			finalHeader, err := e.headers.ByBlockID(e.finalizedBlockTracker.NewestBlock().BlockID)
-			if err != nil { // no expected errors
-				ctx.Throw(err)
-			}
-			e.core.ProcessFinalizedBlock(finalHeader)
-		}
+func (e *Engine) handleFinalizedBlock(block *model.Block) error {
```

Comment:
Following up on my [previous comment](), I would strongly prefer a naming very close to the interface-specified consumer method  [`OnFinalizedBlock`](https://github.com/onflow/flow-go/blob/a6f6fd591aa6d776b50ec47c6d83c79461a9eb20/consensus/hotstuff/consumer.go#L28-L33)



```suggestion
// onFinalizedBlock informs compliance.Core about finalization of the respective block.
// The input to this callback is treated as trusted. This method should be executed on
// `OnFinalizedBlock` notifications from the node-internal consensus instance.
// No errors expected durning normal operations.
func (e *Engine) onFinalizedBlock(block *model.Block) error {
```